### PR TITLE
Fail the build when the user's script fails

### PR DESCRIPTION
**Pull request by robrich**

### CODE REVIEW

File: Tasks/AzurePowerShell/RunAzurePowerShell.ps1 [hunk 1]
```diff
@@ -20,5 +20,8 @@ Write-Host "ScriptPath= $ScriptPath"
 $scriptCommand = "& `"$ScriptPath`" $scriptArguments"
 Write-Host "scriptCommand= $scriptCommand"
 Invoke-Expression -Command $scriptCommand
+if ($LASTEXITCODE -ne 0) {
+    Write-Error -Message "##[error]BUILD FAILED: $LASTEXITCODE"
```
Comment (chrisrpatterson) [inline Tasks/AzurePowerShell/RunAzurePowerShell.ps1]: you don't need the ##[error] here we have overridden the Write-Error so it should take care of the formatting for you for the log highlighting.

Comment (robrich) [inline Tasks/AzurePowerShell/RunAzurePowerShell.ps1]: I agree.  Should we even include "BUILD FAILED"?

**Comment by bryanmacfarlane** [comment]
Muthu, can you work with Eric.  This has been rewritten and we just can't take it like it is.  It will break previous behavior.  We have to add an input to influence behavior and the default would be the previous behavior.

I believe this is irrelevant since it has been rewritten since.

**Comment by mvvsubbu** [comment]
Closing this PR as this change is no more relevant

**Pull request closed without merging**